### PR TITLE
🏷️ Implement the ForumCategoryList component

### LINKED ISSUE #1065: Forum category component
@[figma](https://www.figma.com/file/GlgN8uBRtvtMJtiOsdtDF7/Pioneer-Design?node-id=4409%3A129866)

Some date might be loaded after the inital load – the component might accept it as an optional prop or a sub component that displays "loading".

Loaded by general query:
- [x] Name
- [x] Description
- [ ] moderators ids
- [ ] total threads (if available in the query node)

Loaded asynchronously:
- [ ] moderators avatars
- [ ] last post
- [ ] popular post

### PR DESCRIPTION
**Pull request by thesan**
Closes #1065

### CODE REVIEW

File: packages/ui/src/forum/types/ForumCategory.ts [hunk 1]
```diff
@@ -3,16 +3,25 @@ import { ForumCategoryFieldsFragment } from '@/forum/queries/__generated__/forum
...
+  avatar?: string
 }
 
 interface ForumSubCategory {
   id: string
   title: string
 }
 
-export const asForumCategory = (fields: ForumCategoryFieldsFragment): ForumCategory => ({
+export const asForumCategory = (fields: Omit<ForumCategoryFieldsFragment, '__typename'>): ForumCategory => ({
```
Comment (jodator) [inline packages/ui/src/forum/types/ForumCategory.ts:21]: :+1:

File: packages/ui/src/forum/components/category/storybook-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+export const asModerator = (hasHandle = true, hasAvatar = false) => {
+  return (index: number): ForumModerator => ({
+    id: String(index),
+    membershipId: '0',
+    handle: hasHandle ? faker.name.firstName() : undefined,
+    avatar: hasAvatar ? faker.image.avatar() : undefined,
+  })
+}
+
+export const asPost = (text: string): ForumPost | undefined => {
```
Comment (jodator) [inline packages/ui/src/forum/components/category/storybook-helpers.ts:17]: Maybe those helpers should be named a bit differently to not confuse them with standard `as...()` casting functions.

I'd go with something like `asStorybookPost()` & `asStorybookModerator()` or only `asStoryPost()`, WDYT?

File: packages/ui/src/forum/components/category/storybook-helpers.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+  if (text)
+    return {
+      id: '0',
+      createdAt: sub(Date.now(), { minutes: 25 }).toISOString(),
+      createdAtBlock: asBlock(),
```
Comment (jodator) [inline packages/ui/src/forum/components/category/storybook-helpers.ts:22]: There should be a helper now for this. I've created something like `generateRandomBlock()` so you can find this and re-create for storybook purposes.

**Review by jodator — APPROVED** [review]
Looks good 👌 I've added some suggestions to add, so please check them together with conflicts.

File: packages/ui/src/forum/components/category/storybook-helpers.ts (hunk 2 above)
Comment (thesan) [reply packages/ui/src/forum/components/category/storybook-helpers.ts:17]: Thanks, sorry I had not seen the review while dealing with the conflict. Your right I will rename these

**Pull request merged**